bot/handlers/sending_videos/compile_selected_clips_handler.py

```diff
@@ -31,50 +29,57 @@ def __init__(self, message: str) -> None:
             super().__init__(self.message)
 
     def get_commands(self) -> List[str]:
-        return ["polaczklipy", "concatclips", "pk"]
+        return ["połączklipy", "polaczklipy", "concatclips", "pk"]
 
     async def _do_handle(self, message: Message) -> None:
         content = message.text.split()
 
         if len(content) < 2:
             return await self._reply_invalid_args_count(message, get_invalid_args_count_message())
 
-        clip_names = content[1:]
-        selected_clips_data = await self.__get_selected_clips_data(clip_names, message.from_user.username, message)
+        try:
+            clip_numbers = [int(clip) for clip in content[1:]]
+        except ValueError:
+            return await self._reply_invalid_args_count(message, get_invalid_args_count_message())
+
+        user_clips = await DatabaseManager.get_saved_clips(message.from_user.id)
 
-        if not selected_clips_data:
+        selected_clips = []
+        for clip_number in clip_numbers:
+            if 1 <= clip_number <= len(user_clips):
+                selected_clips.append(user_clips[clip_number - 1])
+            else:
+                return await self._reply_invalid_args_count(message, get_invalid_args_count_message())
+
+        if not selected_clips:
             return await self.__reply_no_matching_clips_found(message)
 
         selected_segments = []
-        for clip_data, duration in selected_clips_data:
+        for clip in selected_clips:
             temp_file = tempfile.NamedTemporaryFile(delete=False, delete_on_close=False, suffix=".mp4")
-            temp_file.write(clip_data)
+            temp_file.write(clip.video_data)
             temp_file.close()
             selected_segments.append({
                 "video_path": temp_file.name,
                 "start": 0,
-                "end": duration,
+                "end": clip.duration,
             })
 
+        total_duration = sum(clip.duration for clip in selected_clips)
+
+        if not await DatabaseManager.is_admin_or_moderator(message.from_user.id) and total_duration > settings.MAX_CLIP_DURATION:
+            await message.answer(get_clip_time_message())
+            return
+
+        if not await DatabaseManager.is_admin_or_moderator(message.from_user.id) and len(selected_segments) > settings.MAX_CLIPS_PER_COMPILATION:
+            await message.answer(get_max_clips_exceeded_message())
+            return
```

Comment:
sprawdz najpierw pierwszy warunek a potem zagniezdz ify z tym drugim, bo sprawdzasz to samo 2 razy, no i mozna tez to samo z tym is any validation failed

Reply:
is any validation failed

w zasadzie to można w każdym handlarze dać 

to zmienić to wszędzie ?

Reply:
w sumie, mozesz zrobic cos takiego, ze w klasie bazowej dodasz pure virtual metode o tej nazwie i wykonasz ja zanim wejdziesz w handle, a klasy childy beda ja implementowac

Reply:
dobra done tylko nwm co z tym zrobić bo dwie mają taka sama ta implementacje to co do utils dać ? czy gdzieś do klasy bazowej ?



************* Module bot.video.utils
bot\video\utils.py:1:0: R0801: Similar lines in 2 files
==bot.handlers.sending_videos.compile_clips_handler:[42:52]
==bot.handlers.sending_videos.select_clip_handler:[31:41]
        content = message.text.split()
        if len(content) < 2:
            await self._reply_invalid_args_count(message, get_invalid_args_count_message())
            return True
        return False

    async def _do_handle(self, message: Message) -> None:
        content = message.text.split()

        last_search = await DatabaseManager.get_last_search_by_chat_id(message.chat.id) (duplicate-code)
bot\video\utils.py:1:0: R0801: Similar lines in 2 files
==bot.handlers.sending_videos.compile_clips_handler:[42:51]
==bot.handlers.sending_videos.compile_selected_clips_handler:[34:43]
        content = message.text.split()
        if len(content) < 2:
            await self._reply_invalid_args_count(message, get_invalid_args_count_message())
            return True
        return False

    async def _do_handle(self, message: Message) -> None:
        content = message.text.split()
 (duplicate-code)

Reply:
`is_any_validation_failed` powinno być wirtualne w klasie bazowej, a childy implementować.

Co do tego konkretnego duplikatu ja bym zrobił tak.

1) W klasie bazowej funkcja wirutalna `get_expected_args_count_range`, w tym wypadku zwraca `(2)`, ale może być np. `(2,3)`
2) W klasie bazowej funkcja sprawdzająca czy `len(message.text.split()) in get_expected_args_count_range`
3) Wywalenie tych walidacji ze wszystkich klas childowych 
4) Zintegrowanie tego z resztą walidacji
5) [optional] Może po rpostu domyślna implementacja `is_any_validation_failed` będzie zawierać wywołanie tej funkcji z punktu 2

Reply:
dobra to zamykam bo robie ten motyw z tym __get_validator_functions